src/WSRequestHandler_ComponantsManagement.cpp

```
@@ -0,0 +1,128 @@
+	
+	obs_source_t* source = obs_source_create(typeIDName, sourceName, newsourceSettings, hotkeyData);
+	obs_source_set_enabled(source, true);
+
+	OBSDataAutoRelease response = obs_data_create();
+	obs_data_set_string(response, "sourceName", obs_source_get_name(source));
+	obs_data_set_string(response, "sourceType", obs_source_get_id(source));
+	obs_data_set_obj(response, "sourceSettings", newsourceSettings);
+	obs_data_set_obj(response, "hotkeyData", hotkeyData);
+	return req->SendOKResponse(response);
+}
+
+/**
+* Add New Scene
+*
+* @param {String} `sceneName` Name of the Scene to be created
+*
+* @api requests
+* @name AddNewScene
+* @category sources
+*/
+HandlerResponse WSRequestHandler::HandleAddNewScene(WSRequestHandler* req)
+{
+	if (!req->hasField("sourceName") || !req->hasField("typeIDName") || !req->hasField("sourceSettings")) {
+		return req->SendErrorResponse("missing request parameters");
+	}
+	const char* sceneName = obs_data_get_string(req->data, "sceneName");
+	obs_scene_create(sceneName);
+	OBSDataAutoRelease response = obs_data_create();
+	obs_data_set_string(response, "sceneName", sceneName);
+	return req->SendOKResponse(response);
+}
+
+/**
+* Add Active Child
+*
+* @param {String} `sourceParentName` Name of the parent source
+* @param {String} `sourceChildName` Name of the child source
+*
+* @api requests
+* @name AddActiveChild
+* @category sources
+*/
+HandlerResponse WSRequestHandler::HandleAddActiveChild(WSRequestHandler* req)
+{
+	if (!req->hasField("sourceParentName") || !req->hasField("sourceChildName")) {
+		return req->SendErrorResponse("missing request parameters");
+	}
+	const char* sourceParentName = obs_data_get_string(req->data, "sourceParentName");
+	const char* sourceChildName = obs_data_get_string(req->data, "sourceChildName");
+	obs_source_t* sourceParent= obs_get_source_by_name(sourceParentName);
+	obs_source_t* sourceChild = obs_get_source_by_name(sourceChildName);
+	
+	obs_source_add_active_child(sourceParent, sourceChild);
```

Please return an error if this call returns `false`.